controllers/machinehealthcheck_controller.go

```diff
@@ -269,7 +310,27 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 			t.string(),
 		)
 	}
+
 	for _, t := range healthy {
+
+		if t.Machine.Status.RemediationRef != nil {
+			ref := t.Machine.Status.RemediationRef
+			t.Machine.Status.RemediationRef = nil // remove remediation reference
+
+			// Get remediation request object
+			obj, err := external.Get(ctx, r.Client, ref, t.Machine.Namespace)
+			if err != nil && !apierrors.IsNotFound(errors.Cause(err)) {
+				return ctrl.Result{}, errors.Wrapf(err, "Failed to fetch remediation request")
+			}
+
+			if obj != nil {
+				// Issue a delete for remediation request.
+				if err := r.Client.Delete(ctx, obj); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Perhaps we should check first if `obj` has no `DeletionTimestamp` to avoid hot loop on this one, in case that the external remediation controller (or something else) adds a finalizer